pkg/server/server.go

```diff
@@ -1369,6 +1380,12 @@ func (s *Server) PreStart(ctx context.Context) error {
 		return errors.Wrap(err, "invalid init state")
 	}
 
+	// Apply any cached initial settings (and start the gossip listener) as early
+	// as possible, to avoid spending time with stale settings.
```

Comment:
> We could consider returning the settings from the Join RPC (which replaces gossip-join), but let's hold off on that, at least in this PR, though it would be a good follow-up.

I assume this suggestion was motivated by not wanting to spend any time with stale settings? Instead of serializing settings through the Join RPC (which I think should stay simple - just node + cluster id + version, as it is today), perhaps what we could do here instead is wait for a gossip update containing the up-to-date settings. We'll want to do this after having joined the cluster, after having started up gossip, but before having fully started up the node. What do you think?